medicai/trainer/nnunet/planning/planners.py

```
@@ -0,0 +1,761 @@
+  - 2D plan  (median slice shape)
+  - 3D full-res plan  (median volume at target spacing)
+  - 3D low-res plan  (if cascade is triggered)
+  - Cascade decision (volume too large → low-res + full-res)
+  - Patch sizes, batch sizes, pooling depths
+  - Normalization schemes per modality
+
+Design follows the original nnU-Net ExperimentPlanner logic without
+hardcoding per-dataset constants.
+"""
+
+from __future__ import annotations
```

Ruff (0.15.9)

[error] `from __future__` imports must occur at the beginning of the file

(F404)